core/deployment/src/main/java/io/quarkus/deployment/ApplicationArchive.java

```diff
@@ -26,33 +29,72 @@
      * jar, but rather a path to the root of the mounted {@link com.sun.nio.zipfs.ZipFileSystem}
      *
      * @return The archive root.
+     * @deprecated in favor of {@link #getRootDirs()}
      */
+    @Deprecated
     Path getArchiveRoot();
 
     /**
      *
      * @return <code>true</code> if this archive is a jar
+     * @deprecated does not appear to be used anywhere and now it shouldn't be
      */
+    @Deprecated
     boolean isJarArchive();
 
     /**
      * If this archive is a jar file it will return the path to the jar file on the file system,
      * otherwise it will return the directory that this corresponds to.
+     *
+     * @deprecated in favor of {@link #getPaths()}
      */
+    @Deprecated
     Path getArchiveLocation();
 
+    /**
+     *
+     * Returns paths representing the archive root directories. Note that every path in this collection
+     * is guaranteed to be a directory. If the actual application archive appears to be a JAR,
+     * this collection will include a path to the root of the mounted {@link java.nio.file.FileSystem}
+     * created from the JAR.
+     *
+     * @return The archive root directories.
+     */
+    PathsCollection getRootDirs();
```

Comment:
What's the benefit of having `PathsCollection` over `List<Path>` ?

Reply:
the utility methods I suppose, like `resolveExistingOrNull`?

Reply:
Yes, it allows/will allow us to add functionality we find useful in our context.